ui/hooks/gasFeeInput/useTransactionFunction.test.js

```diff
@@ -0,0 +1,132 @@
+import { renderHook } from '@testing-library/react-hooks';
+
+import {
+  CUSTOM_GAS_ESTIMATE,
+  EDIT_GAS_MODES,
+  GAS_RECOMMENDATIONS,
+} from '../../../shared/constants/gas';
+import * as Actions from '../../store/actions';
+import {
+  FEE_MARKET_ESTIMATE_RETURN_VALUE,
+  configureEIP1559,
+} from './test-utils';
+import { useTransactionFunctions } from './useTransactionFunctions';
+
+jest.mock('../useGasFeeEstimates', () => ({
+  useGasFeeEstimates: jest.fn(),
+}));
+
+jest.mock('react-redux', () => {
+  const actual = jest.requireActual('react-redux');
+
+  return {
+    ...actual,
+    useSelector: jest.fn(),
```

Comment:
Not sure how that can be done, `jest.mock` seems to allow mocking modules only.

Reply:
Right. Instead of mocking `useSelector`, mock the functions that you pass to `useSelector`.

Although now that I look at `useTransactionFunction`, it doesn't appear that you use `useSelector`. So I'm not sure you need to mock `useSelector`. But the same thing applies to `useDispatch` — mock the functions you pass to `dispatch`. For instance:

``` javascript
import {
  createCancelTransaction,
  createSpeedUpTransaction,
  updateCustomSwapsEIP1559GasParams,
  updateSwapsUserFeeLevel,
  updateTransaction as updateTransactionFn,
} from '../../store/actions';

jest.mock('../../store/actions', () => {
  return {
    createCancelTransaction: jest.fn(),
    createSpeedUpTransaction: jest.fn(),
    updateCustomSwapsEIP1559GasParams: jest.fn(),
    updateSwapsUserFeeLevel: jest.fn(),
    updateTransaction: jest.fn(),
  }
});
```

Now when `useTransactionFunctions` calls e.g. `dispatch(createCancelTransaction(...))` then the mock version of `createCancelTransaction` will be used.

Reply:
Good point, I just fixed